primer/brand
# Encapsulate focus within `dialog`

## Summary



Adds a focus-trap to the search `dialog`. 

Resolves `dialog` issue from https://github.com/github/primer/issues/1668 (internal).

## List of notable changes:



- Pulls in Primer `useFocusTrap` hook to encapsulate user focus within `dialog`.

## What should reviewers focus on?

- Ensure that focus does not leave dialog unless dismissed manually (currently through the close button)

## Steps to test:



1. Open up Storybook example of `SubdomainNavbar` with search feature present.
2. See if focus is trapped within the `dialog` after activation of the search button.

## Supporting resources (related issues, external links, etc):

- [`useFocusTrap`](https://github.com/primer/react/blob/main/src/hooks/useFocusTrap.ts)

## Contributor checklist:


## Reviewer checklist:
